ompi/mca/mtl/ofi/mtl_ofi_component.c

```diff
@@ -411,6 +429,76 @@ ompi_mtl_ofi_component_init(bool enable_progress_threads,
         goto error;
     }
 
+    /* Check if FI_REMOTE_CQ_DATA is supported, if so send the source rank there
+     * FI_DIRECTED_RECV is also needed so receives can discrimate the source
+     */
+    prov_tmp_name = strdup(prov->fabric_attr->prov_name);
+    if(!prov_tmp_name){
+        goto error;
+    }
+
+    free(hints->fabric_attr->prov_name);
+    hints->fabric_attr->prov_name=prov_tmp_name;
+    hints->caps = FI_TAGGED | FI_DIRECTED_RECV;
+    /* Ask for the size that OMPI uses for the source rank number */
+    hints->domain_attr->cq_data_size = sizeof(int);
+    ret = fi_getinfo(fi_version, NULL, NULL, 0ULL, hints, &prov_cq_data);
+
+   if (OPAL_UNLIKELY(0 != ret) && (-FI_ENODATA != ret)) {
+        opal_show_help("help-mtl-ofi.txt", "OFI call fail", true,
+                       "fi_getinfo",
+                       ompi_process_info.nodename, __FILE__, __LINE__,
+                       fi_strerror(-ret), -ret);
+        goto error;
+    }
+   else if ((NULL != prov_cq_data) &&
```

Comment:
Open MPI style is `} else if (....` on one line.  So the `}` on line 453 should be on line 454.

Also, why an else if, if your previous test is going to goto NULL.  It's a little defensive, but it's also a bit hard to follow.

Reply:
Will more the `else if` to line 453. 
The `else if` is to check if  the provider indeed supports CQ_DATA.  fi_getinfo can return 0 and find no provider.